src/ctap/hid/mod.rs

```diff
@@ -146,16 +146,13 @@ impl CtapHid {
 
     // Process an incoming USB HID packet, and optionally returns a list of outgoing packets to
     // send as a reply.
-    pub fn process_hid_packet<R, CheckUserPresence>(
+    pub fn process_hid_packet(
```

Comment:
I'm thinking where everything should live. Not actionable for now, but I'm looking into the refactoring of HID right now. To let you know about my ideas: And since most of the code in this crate is actually independent of `CtapState` and `ctap1`, maybe there is a good way to move all of `hid` out of the `ctap` mod.

Reply:
Yes, there's definitely a lot of refactoring that can/should be done. This PR only starts to incrementally address the high-level picture of the library.